### ISSUE TITLE
C++ Unit Tests cause build to fail

### ISSUE THREAD

**Issue by mcm001** [wpilibsuite/vendor-template#3]
As I discussed with ThadHouse on the Discord in February, C++ unit tests in a vendor library with JNI stuff causes the build to fail. The workaround is to disable C++ unit tests, but that's not a great workaround as I would like to be able to also test C++ code. This is what Thad said in Feb:

```
Thats a bug. I don't have a workaround for it.
[Allwpilib fixes it], in a very hacky way.
I can get it into the vendor build, but it requires custom setup.
But I want to fix it the right way. Because the fix we have is both a speed perf hack and a hack for fixing that issue.
Vendors don't need the speed perf hack, its very hacky but I would likely to actually fix that issue.
Basically, gtest in gradle doesn't use the library, but actually builds another copy of the library directly.
And its that extra build that doesn't know about JNI, so it fails.
The hack solution is to disable the test separate build, and actually test against the built shared library.
Which solves the linking issue and also removes an entire copy of the build.
```

**Comment by ThadHouse** [comment]
Fixed in https://github.com/wpilibsuite/vendor-template/commit/29ba09947c51ef51942e61f5c895e6620dfae16c
